sktime/forecasting/compose/_ensemble.py

```diff
@@ -7,16 +7,181 @@
+        feature importance scores (decision tree-based models). If None, then
+        a GradientBoostingRegressor(max_depth=5) is used. The regressor can also be a
+        sklearn.Pipeline().
+    test_size : int or float, optional, default=None
+        Used to do an internal temporal_train_test_split(). The test_size data
+        will be the endog data of the regressor and it is the most recent data.
+        The exog data of the regressor are the predictions from the temporarily
+        trained ensemble models. If None, it will be set to 0.25.
+    n_jobs : int or None, optional, default=None
+        The number of jobs to run in parallel for fit. None means 1 unless
+        in a joblib.parallel_backend context.
+        -1 means using all processors.
+
+    Attributes
+    ----------
+    regressor_ : sklearn-like regressor
+        Fitted regressor.
+    weights_ : np.array
+        The weights based on either regressor.feature_importances_ or
+        regressor.coef_ values.
+
+    See Also
+    --------
+    EnsembleForecaster
+
+    Example
+    -------
+    >>> from sktime.forecasting.compose import AutoEnsembleForecaster
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> from sktime.forecasting.trend import PolynomialTrendForecaster
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> forecasters = [("trend", PolynomialTrendForecaster()),\
+                        ("naive", NaiveForecaster())]
+    >>> forecaster = AutoEnsembleForecaster(forecasters=forecasters, n_jobs=2)
+    >>> forecaster.fit(y=y, X=None, fh=[1,2,3])
+    AutoEnsembleForecaster(...)
+    >>> y_pred = forecaster.predict()
+    """
+
+    _required_parameters = ["forecasters"]
+    _tags = {
+        "univariate-only": False,
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+    }
+
+    def __init__(
+        self,
+        forecasters,
+        regressor=None,
+        test_size=None,
+        n_jobs=None,
+    ):
+        super(AutoEnsembleForecaster, self).__init__(
+            forecasters=forecasters, n_jobs=n_jobs
+        )
+        self.regressor = regressor
+        self.test_size = test_size
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit to training data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            Target time series to which to fit the forecaster.
+        fh : int, list or np.array, optional, default=None
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, optional, default=None
+            Exogenous variables are ignored.
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        _, forecasters = self._check_forecasters()
+        self._check_regressor()
+
+        # get training data for meta-model
+        if X is not None:
+            y_train, y_test, X_train, X_test = temporal_train_test_split(
```

Comment:
hm, do we want to upgrade `temporal_train_test_split` so it returns `None` for the `X`-type returns if `X` is `None`? Would make this code (and probably other code) easier.

Reply:
ah, that wouldn't work, since it would break the expected two arguments